File: 01_projects/code/python-playwright/cbaps-dawms/basic-pom/cbaps/pages/routing_approval_page.py
"""
RoutingApprovalPage: Route requisition to approvers
Select approver and submit routing
"""

import logging
from playwright.sync_api import Page, Locator
from cbaps.library.playwright_manager import PlaywrightManager

logger = logging.getLogger(__name__)


class RoutingApprovalPage:
    """
    RoutingApprovalPage handles approval routing workflow.
    This is the Python equivalent of the Java RoutingApprovalPage.
    """
    
    def __init__(self, page: Page, pwm: PlaywrightManager):
        """
        Initialize RoutingApprovalPage
        
        Args:
            page: Playwright Page object
            pwm: PlaywrightManager instance
        """
        self.page = page
        self.pwm = pwm
        
        # Locators
        self.select_approver_dropdown: Locator = page.locator("#approver")
        self.submit_routing_button: Locator = page.locator(
            "button:has-text('Submit Routing')"
        )
        
        # Stability anchor: wait for routing page to load
        pwm.wait_visible("text=Routing")
        logger.info("Routing Approval Page loaded successfully.")
    
    def submit_for_approval(self, approver: str) -> 'StatusTrackerPage':
        """
        Submit requisition for approval to selected approver
        
        Args:
            approver: Approver name/role
            
        Returns:
            StatusTrackerPage instance
        """
        from cbaps.pages.status_tracker_page import StatusTrackerPage
        
        self.select_approver_dropdown.select_option(approver)
        logger.info(f"Selected approver: {approver}")
        
        self.pwm.click(self.submit_routing_button)
        logger.info("Clicked 'Submit Routing' button.")
        
        return StatusTrackerPage(self.page, self.pwm)

File: 01_projects/code/java-playwright/cbaps-dawms/basic-pom/src/test/java/com/cbaps/pages/RoutingApprovalPage.java
package com.cbaps.pages;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import com.microsoft.playwright.Locator;
import com.microsoft.playwright.Page;
import com.cbaps.library.PlaywrightManager;

/**
 * RoutingApprovalPage: Route requisition to approvers
 * Select approver and submit routing
 */
public class RoutingApprovalPage {

	private static final Logger log = LoggerFactory.getLogger(RoutingApprovalPage.class);
	private final Page page;
	private final PlaywrightManager pwm;

	private final Locator selectApproverDropdown;
	private final Locator submitRoutingButton;

	public RoutingApprovalPage(Page page, PlaywrightManager pwm) {
		this.page = page;
		this.pwm = pwm;

		// Initialize locators
		this.selectApproverDropdown = page.locator("#approver");
		this.submitRoutingButton = page.locator("button:has-text('Submit Routing')");

		// Stability anchor: wait for routing page to load
		pwm.waitVisible("text=Routing");
		log.info("Routing Approval Page loaded successfully.");
	}

	/**
	 * Submit requisition for approval to selected approver
	 * @param approver - Approver name/role
	 * @return StatusTrackerPage
	 */
	public StatusTrackerPage submitForApproval(String approver) {
		selectApproverDropdown.selectOption(approver);
		log.info("Selected approver: " + approver);
		
		pwm.click(submitRoutingButton);
		log.info("Clicked 'Submit Routing' button.");
		
		return new StatusTrackerPage(page, pwm);
	}
}
